### PR TITLE
Add API call to get minidisks list

### PR DESCRIPTION
**Pull request by Bischoff**
This PR adds missing GET operation for the list of minidisks of a guest:
```
GET /guests/{userid}/disks
```

This PR solves issue #796.

It contains changes to:
 * zvmsdk
 * zvmconnector
 * unit tests
 * documentation

### CODE REVIEW

File: zvmsdk/smtclient.py [hunk 1]
```diff
@@ -387,6 +387,47 @@ def get_adapters_info(self, userid):
...
+            results = self._request(rd)
+        ret = results['response']
+        disks_info = []
+        for line in ret:
+            if line.startswith('MDISK'):
+                vdev = line.split(' ')[1]
+                rd = ' '.join((
+                    "SMAPI %s API Image_Disk_Query" % userid,
+                    "--operands",
+                    "-k 'vdasd_id=%s'" % vdev))
```
Comment (bjhuangr) [inline zvmsdk/smtclient.py:404]: why not using `-k vdasd_id=all` to get all guest's minidisk info in one `Image_Disk_Query` call ?

**Review by bjhuangr — COMMENTED** [review]
Another quesion, how to deal with temporary linked MDISK (not owned by this guest vm). Whether such linked MDISK info should be included in the output of the new API.

File: zvmsdk/smtclient.py (hunk 1 above)
Comment (Bischoff) [reply zvmsdk/smtclient.py:404]: I can try, yes, thanks for the idea.

**Comment by Bischoff** [comment]
> Another quesion, how to deal with temporary linked MDISK (not owned by this guest vm). Whether such linked MDISK info should be included in the output of the new API.

We could have a flag wether to include them or not. Or just always include them. But the question is whether it would be cleaner to have a different call for them...

Admitting we decide to do it, how could I access that information on the SMAPI side?

**Comment by bjhuangr** [comment]
The same API `Image_Disk_Query` is able to query MDISKs that linked from other guest.

I'm fine to add a flag for include/exclude the disks not owned by the target guest.

File: zvmsdk/smtclient.py (hunk 1 above)
Comment (Bischoff) [reply zvmsdk/smtclient.py]: Done. It is indeed simpler.

**Comment by Bischoff** [comment]
> The same API `Image_Disk_Query` is able to query MDISKs that linked from other guest.

Done, it works fine too.


> I'm fine to add a flag for include/exclude the disks not owned by the target guest.

I don't see the way to distinguish between local or linked, neither in input nor in output.

Here is what the API call returns:
```
 DASD VDEV: 0100
   RDEV: 6018
   Access type: R/W
   Device type: 3390
   Device size: 17477
   Device units: Cylinders
   Device volume label: VM6018
 
 DASD VDEV: 0190
   RDEV: 6000
   Access type: R/O
   Device type: 3390
   Device size: 214
   Device units: Cylinders
   Device volume label: M01RES
 
 DASD VDEV: 019D
   RDEV: 6000
   Access type: R/O
   Device type: 3390
   Device size: 292
   Device units: Cylinders
   Device volume label: M01RES
 
 DASD VDEV: 019E
   RDEV: 6000
   Access type: R/O
   Device type: 3390
   Device size: 500
   Device units: Cylinders
   Device volume label: M01RES
 
 DASD VDEV: 0592
   RDEV: 6000
   Access type: R/O
   Device type: 3390
   Device size: 240
   Device units: Cylinders
   Device volume label: M01RES

```

**Comment by bjhuangr** [comment]
Is there a customer requirement that rely on this new API?
If this is not the case, adding this API just because we missed this part. Then I suggest we can return all guest vm awared disks information at first.

File: zvmsdk/smtclient.py [hunk 2]
```diff
@@ -387,6 +387,41 @@ def get_adapters_info(self, userid):
...
+                        disk['device_type'] = value
+                    elif key == 'Device size':
+                        disk['device_size'] = int(value)
+                    elif key == 'Device units':
+                        disk['device_units'] = value
+                    elif key == 'Device volume label':
+                        disk['volume_label'] = value
+                else:
+                    disks_info.append(disk)
+                    disk = dict()
```
Comment (bjhuangr) [inline zvmsdk/smtclient.py:422]: the logic rely on `always have a blank line at the end of the SMAPI output`, please double check this. Make sure the feilong API can get all disks info as expected.

Comment (Bischoff) [reply zvmsdk/smtclient.py:422]: There is a blank line at the end of SMAPI output, at least when I test.

I double-checked that we get the last disk.

**Comment by bjhuangr** [comment]
I meant, we can start from easy approach - return all disks info.

**Comment by Bischoff** [comment]
> I meant, we can start from easy approach - return all disks info.

Of course. That's the current state of this PR.

What I can do later on is to add a boolean field to the returned structures: directly connected yes/no. That way, we would keep an ascending compatibility.

**Review by bjhuangr — APPROVED** [review]

**Pull request merged**